File: onyx-database-tests/src/entities/PerformanceEntityChild.java
package entities;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import com.onyx.persistence.IManagedEntity;
import com.onyx.persistence.annotations.*;

/**
 * Created by timothy.osborn on 1/13/15.
 */
@Entity
@JsonIdentityInfo(generator = ObjectIdGenerators.IntSequenceGenerator.class, property = "@id")
public class PerformanceEntityChild extends AbstractEntity implements IManagedEntity
{
    @Identifier(generator = IdentifierGenerator.SEQUENCE)
    @Attribute
    public Long id;

    @Attribute
    public String someOtherField;

    @Relationship(type = RelationshipType.ONE_TO_ONE, inverseClass = PerformanceEntity.class, inverse = "child")
    public PerformanceEntity parent;

}

File: onyx-database-tests/src/main/kotlin/entities/PerformanceEntityChild.kt
package entities

import com.fasterxml.jackson.annotation.JsonIdentityInfo
import com.fasterxml.jackson.annotation.ObjectIdGenerators
import com.onyx.persistence.IManagedEntity
import com.onyx.persistence.annotations.*
import com.onyx.persistence.annotations.values.IdentifierGenerator
import com.onyx.persistence.annotations.values.RelationshipType

/**
 * Created by timothy.osborn on 1/13/15.
 */
@Entity
@JsonIdentityInfo(generator = ObjectIdGenerators.IntSequenceGenerator::class, property = "@id")
class PerformanceEntityChild : AbstractEntity(), IManagedEntity {
    @Identifier(generator = IdentifierGenerator.SEQUENCE)
    @Attribute
    var id: Long? = null

    @Attribute
    var someOtherField: String? = null

    @Relationship(type = RelationshipType.ONE_TO_ONE, inverseClass = PerformanceEntity::class, inverse = "child")
    var parent: PerformanceEntity? = null
}
